Cython/Compiler/Symtab.py

```diff
@@ -2150,6 +2151,12 @@ def needs_tp_clear(self):
         """
         return self.needs_gc() and not self.directives.get('no_gc_clear', False)
 
+    def mangle_class_private_name(self, name):
+        # a few utilitycode names need to specifically be ignored
+        if name and name.startswith("__pyx_unpickle_"):
+            return name
```

Comment:
I'm not so happy that we have to look at the Python name here (a cname would be safer), but at least for now, I think that's acceptable.

Reply:
Unfortunately I've had to expand this further to get memoryviews to work. I'm trying to think if there's an easy way to detect it - for memoryviews we could look up through the parents scopes to see if one is a `NonManglingModuleScope`, but the pickle code isn't generated in one of these so would be harder to detect.

Reply:
And cname isn't usually available until you've found the entry, so that isn't really an option.

Reply:
> look up through the parents scopes to see if one is a `NonManglingModuleScope`

Either that, or the `NonManglingModuleScope` could do something that impacts its child scopes.

Reply:
(But as I said, excluding `__pyx_…` is probably good enough for now, and for this rather special feature.)

Reply:
Another alternative might be an (probably?) undocumented directive that turns it off. But I agree this is OK for now.